app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeScreen.kt

```diff
@@ -79,53 +79,80 @@ fun WelcomeScreen(viewModel: WelcomeViewModel = hiltViewModel()) {
     WelcomeContent(viewModel)
 }
 
-
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 private fun WelcomeContent(viewModel: WelcomeViewModel) {
-    Scaffold(modifier = Modifier.padding(vertical = MaterialTheme.wireDimensions.welcomeVerticalPadding)) { internalPadding ->
-        Column(
-            horizontalAlignment = Alignment.CenterHorizontally,
-            verticalArrangement = Arrangement.SpaceBetween,
-            modifier = Modifier.padding(internalPadding)
-        ) {
-            Icon(
-                imageVector = ImageVector.vectorResource(id = R.drawable.ic_wire_logo),
-                tint = MaterialTheme.colorScheme.onBackground,
-                contentDescription = stringResource(id = R.string.content_description_welcome_wire_logo)
-            )
+    ConstraintLayout {
+        val (closeIcon, welcomeScreen) = createRefs()
+
+        if (viewModel.isThereActiveSession && viewModel.state.isOnPremises) {
+            Icon(painter = painterResource(id = R.drawable.ic_close),
+                contentDescription = stringResource(R.string.content_description_right_arrow),
+                modifier = Modifier
+                    .clickable(Clickable(enabled = true, onClick = { viewModel.navigateBack() }))
+                    .padding(horizontal = dimensions().spacing24x, vertical = dimensions().spacing32x)
+                    .constrainAs(closeIcon) {
+                        start.linkTo(parent.start)
+                        top.linkTo(parent.top)
+                    })
+        }
```

Comment:
It's better to keep everything inside the `Scaffold` and just add `topBar` to the `Scaffold`, it can be just the clickable `Icon` or `WireCenterAlignedTopAppBar` with empty title to keep it consistent throughout the whole app